src/Facets/CelerCircleBridgeFacet.sol

```diff
@@ -0,0 +1,104 @@
+/// @custom:version 1.0.0
+contract CelerCircleBridgeFacet is
+    ILiFi,
+    ReentrancyGuard,
+    SwapperV2,
+    Validatable
+{
+    /// Storage ///
+
+    /// @notice The address of the TokenMessenger on the source chain.
+    ICircleBridgeProxy private immutable circleBridgeProxy;
+
+    /// @notice The USDC address on the source chain.
+    address private immutable usdc;
+
+    /// Constructor ///
+
+    /// @notice Initialize the contract.
+    /// @param _circleBridgeProxy The address of the CircleBridgeProxy on the source chain.
+    /// @param _usdc The address of USDC on the source chain.
+    constructor(ICircleBridgeProxy _circleBridgeProxy, address _usdc) {
+        circleBridgeProxy = _circleBridgeProxy;
+        usdc = _usdc;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via CelerCircleBridge
+    /// @param _bridgeData Data containing core information for bridging
+    function startBridgeTokensViaCelerCircleBridge(
+        BridgeData calldata _bridgeData
+    )
+        external
+        nonReentrant
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+        onlyAllowSourceToken(_bridgeData, usdc)
+    {
+        LibAsset.depositAsset(usdc, _bridgeData.minAmount);
+        _startBridge(_bridgeData);
+    }
+
+    /// @notice Performs a swap before bridging via CelerCircleBridge
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    function swapAndStartBridgeTokensViaCelerCircleBridge(
+        BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+        onlyAllowSourceToken(_bridgeData, usdc)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData);
+    }
+
+    /// Private Methods ///
+
+    /// @dev Contains the business logic for the bridge via CelerCircleBridge
+    /// @param _bridgeData The core information needed for bridging
+    function _startBridge(BridgeData memory _bridgeData) private {
+        // give max approval for token to CelerCircleBridge bridge, if not already
+        LibAsset.maxApproveERC20(
+            IERC20(usdc),
+            address(circleBridgeProxy),
+            _bridgeData.minAmount
+        );
+
+        // initiate bridge transaction
+        circleBridgeProxy.depositForBurn(
+            _bridgeData.minAmount,
+            uint64(_bridgeData.destinationChainId),
```

Comment:
Maybe we should check before if this value is within uint64 range and throw a meaningful error since gas consumption does not seem to be a worry for this facet